workload.go

```diff
@@ -62,32 +76,59 @@ func (w WorkloadConfig) Config() WorkloadConfig {
 }
 
 func NewWorkloadConfigWithDefaults(c WorkloadConfig) WorkloadConfig {
-	if c.RateControl.OuterLoopRate == 0 {
-		c.RateControl.OuterLoopRate = 50
+	if c.Name == "" {
+		panic("must specify workload Name")
 	}
 
-	if c.Visualization.LatencyHistMin == 0 {
-		c.Visualization.LatencyHistMin = 0
+	if c.WorkloadScale <= 0 || c.WorkloadScale >= 1 {
+		panic("WorkloadScale (if specified) must be greater than 0.0 and less than 1.0")
 	}
 
-	if c.Visualization.LatencyHistMax == 0 {
-		c.Visualization.LatencyHistMax = 50000
+	if c.RateControlConfig.EventRate == 0 {
+		c.RateControlConfig.EventRate = 1000
 	}
 
-	if c.Visualization.LatencyHistSize == 0 {
-		c.Visualization.LatencyHistSize = 1000
+	if c.RateControlConfig.Concurrency == 0 {
+		if c.RateControlConfig.MaxEventRatePerWorker == 0 {
+			c.RateControlConfig.MaxEventRatePerWorker = 100
+		}
+		c.RateControlConfig.Concurrency = int(math.Ceil(c.RateControlConfig.EventRate / float64(c.RateControlConfig.MaxEventRatePerWorker)))
+	} else if c.RateControlConfig.Connections >= c.RateControlConfig.Concurrency {
+		c.DatabaseConfig.ConnectionMultiplier = c.RateControlConfig.Connections / c.RateControlConfig.Concurrency
```

Comment:
I guess we assume that the config is validated in BenchmarkConfig.Validate so this division will always give integer results?